src/utils/request-queue.js

```diff
@@ -20,8 +20,10 @@ class RequestQueue {
       : false; // Defaults to 'false'
 
     // Mark if page environment is unloading
-    helpers.addEventListener('beforeunload', () => {
-      this.pageUnloading = true;
+    helpers.addEventListener('visibilitychange', () => {
+      if (document.visibilityState === 'hidden') {
```

Comment:
`visibilityState=hidden` based on what [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilityState) says can mean the  "document is either a background tab or part of a minimized window, or the OS screen lock is active".

Would it make sense to add in logic to add in some additional logic here to call `send()` if the page becomes visible again? 🤔 

```
if (document.visibilityState === 'visibile') {
    this.send();
}
```

Reply:
Sorry I somehow completely missed the notification for this one. I don't see any reason not to. Good call